src/engine/math_util.c

```diff
@@ -322,6 +322,9 @@ void mtxf_copy(register Mat4 dest, register Mat4 src) {
 
 /// Set mtx to the identity matrix.
 void mtxf_identity(register Mat4 mtx) {
+#ifdef PUPPYPRINT_DEBUG
+    gPuppyCallCounter.matrix++;
+#endif
```

Comment:
small thing that doesn't matter much but maybe these three lines could be made into a macro? since it's repeated a lot and it kinda gets in the way of the vanilla functions otherwise, i think making this a macro would be good

Reply:
followup, maybe the macro could be general use and the field to increment is passed as the value, so for example `matrix` could be passed here, maybe `collision_floor` passed other places, and maybe macros for `u32 first = osGetCount` and `profiler_collision_update(first);` as well